src/System/query_failures.rs

```diff
@@ -0,0 +1,645 @@
+use std::{collections::HashSet, fmt};
+
+use crate::{
+    component::{Component, State},
+    ModelObjects::statepair::StatePair,
+    TransitionSystems::{CompositionType, TransitionSystem, TransitionSystemPtr},
+};
+
+use super::specifics::{SpecificPath, SpecificState};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Composition {
```

Comment:
I think it's a bit confusing having the enum called `Composition`, like the operation, and one of its fields. However, I don't have any good alternatives, maybe `Layout`, `Pattern`, `Configuration`? I'll let you decide what to do